hawkular-bus/hawkular-bus-common/src/main/java/org/hawkular/bus/common/BinaryData.java

```diff
@@ -59,68 +59,52 @@ public void setOnCloseAction(Runnable action) {
         onCloseAction = action;
     }
 
-    public int read() throws IOException {
-        if (unreadInMemoryDataExists()) {
-            return (inMemoryData[inMemoryDataPointer++] & 0xff);
-        } else {
-            return streamData.read();
-        }
-    }
+    public void close() throws IOException {
+        backingStream.close();
```

Comment:
He's talking about the close action. Once we delegate to the close(), we need to call that close action in order to clean up things. This was to clean up the temporary file that was produced when using the binary stream in ActiveMQ but it could be for anything - whether the close succeeded or failed, that needed to happen because there is no other way for the caller to later trigger that close action, hence why we should wrap this in a try..finally:

```java
try {
    backingStream.close();
} finally {
    // if we were asked to do something after close is done, do it now
    if (onCloseAction != null) {
        onCloseAction.run();
    }
}